### PR TITLE
C# patterns reference: example of a positional pattern with identifiers

### PR DESCRIPTION
**Pull request by pkulikov**
Contributes to #20584 
This also fixes the broken links to the code file.
@BillWagner this is ready to merge from my side.
The next PR after this will add introduction and then the article is ready for the final review.

### CODE REVIEW

**Comment by BillWagner** [comment]
closing and reopening for a fresh build

File: docs/csharp/language-reference/operators/patterns.md [hunk 1]
```diff
@@ -96,19 +96,19 @@ Beginning with C# 9.0, you use the `not`, `and`, and `or` pattern combinators to
 
 - *Negation* `not` pattern that matches an expression when the negated pattern doesn't match the expression. For example, you can negate the [constant](#constant-pattern) `null` pattern to check if an expression is non-null, as the following code shows:
 
-  :::code language="csharp" source="snippets/patterns/PatternCombinators.cs" id="NotPattern":::
+  :::code language="csharp" source="snippets/patterns/LogicalPatterns.cs" id="NotPattern":::
 
 - *Conjunctive* `and` pattern that matches an expression when both patterns match the expression. For example, you can combine [relational patterns](#relational-patterns) to check if a value is in a certain range, as the following code shows:
 
-  :::code language="csharp" source="snippets/patterns/PatternCombinators.cs" id="AndPattern":::
+  :::code language="csharp" source="snippets/patterns/LogicalPatterns.cs" id="AndPattern":::
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/patterns.md:103]: This snippet isn't rendering in the review site. Is the snippet ID incorrect?

**Review by BillWagner — CHANGES_REQUESTED** [review]
I like the changes here @pkulikov 

There's one snippet that's not rendering, and once that's fixed, this is ready.

**Review by BillWagner — APPROVED** [review]

File: docs/csharp/language-reference/operators/patterns.md (hunk 1 above)
Comment (pkulikov) [reply docs/csharp/language-reference/operators/patterns.md:103]: Thank you @BillWagner 
The snippet has not been closed. I've committed the fix. Does it render now?

**Comment by BillWagner** [comment]
The build is clean. I'll :shipit: now.

**Pull request merged**